src/main/java/com/google/api/codegen/configgen/ProtoMethodTransformer.java

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.configgen;
+
+import com.google.api.BackendRule;
+import com.google.api.codegen.config.MethodModel;
+import com.google.api.codegen.config.ProtoMethodModel;
+import com.google.api.tools.framework.model.Model;
+
+/** MethodTransformer implementation for proto Methods. */
+public class ProtoMethodTransformer implements MethodTransformer {
+  private static final PagingParameters PAGING_PARAMETERS = new ProtoPagingParameters();
+  private static final String WILDCARD = "*";
+
+  @Override
+  public boolean isIgnoredParameter(String parameter) {
+    return PAGING_PARAMETERS.getIgnoredParameters().contains(parameter);
+  }
+
+  @Override
+  public String getTimeoutMillis(MethodModel method) {
+    Model model = ((ProtoMethodModel) method).getProtoMethod().getModel();
+    for (BackendRule backendRule : model.getServiceConfig().getBackend().getRulesList()) {
+      if (backendRule.getSelector().equals(method.getFullName())) {
+        return String.valueOf((int) Math.ceil(backendRule.getDeadline()));
```

Comment:
This is in seconds in service config, so it needs to be converted to millis for GAPIC config.